lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -93,16 +106,88 @@ defmodule Thrift.Generator.StructBinaryProtocol do
       defp deserialize(<<0, rest::binary>>, acc=%unquote(name){}) do
         {acc, rest}
       end
+
       unquote_splicing(field_deserializers)
+
+      defp deserialize(<<field_type, _id::16-signed, rest::binary>>, acc) do
+        # this is for unknown fields. This can happen if you have an
+        # outdated schema
+        rest
+        |> skip_field(field_type)
+        |> deserialize(acc)
+      end
+
+      unquote(field_skippers)
+    end
+  end
+
+  def field_skippers do
+
+    quote do
+      defp skip_field(<<_value, rest::binary>>, unquote(@bool)), do: rest
+      defp skip_field(<<_value, rest::binary>>, unquote(@byte)), do: rest
+      defp skip_field(<<_::signed-float, rest::binary>>, unquote(@double)), do: rest
+      defp skip_field(<<_::size(16), rest::binary>>, unquote(@i16)), do: rest
+      defp skip_field(<<_::size(32), rest::binary>>, unquote(@i32)), do: rest
+      defp skip_field(<<_::size(64), rest::binary>>, unquote(@i64)), do: rest
+      defp skip_field(<<string_size::32-signed, rest::binary>>, unquote(@string)) do
+        << _::binary-size(string_size), remainder::binary>> = rest
+        remainder
+      end
+
+      # sets and lists
+      defp skip_field(<<elem_type, length::size(32), rest::binary>>, field_type)
+        when field_type == unquote(@set) or field_type == unquote(@list) do
+        skip_list_field(rest, elem_type, length)
+      end
+
+      defp skip_list_field(<<rest::binary>>, _, 0) do
+        rest
+      end
+      defp skip_list_field(<<rest::binary>>, elem_type, remaining) do
+        rest
+        |> skip_field(elem_type)
+        |> skip_list_field(elem_type, remaining - 1)
+      end
+
+      # maps
+      defp skip_field(<<key_type, val_type, length::size(32), rest::binary>>, unquote(@map)) do
+        skip_map_field(rest, key_type, val_type, length)
+      end
+
+      defp skip_map_field(<<rest::binary>>, _, _, 0) do
+        rest
+      end
+      defp skip_map_field(<<rest::binary>>, key_type, val_type, remaining) do
+        rest
+        |> skip_field(key_type)
+        |> skip_field(val_type)
+        |> skip_map_field(key_type, val_type, remaining - 1)
+      end
+
+      defp skip_field(<<rest::binary>>, unquote(@struct)) do
+        skip_struct(rest)
+      end
```

Comment:
All the `skip_field` clauses should be grouped together. Preferably in order of field type ordinal (i.e. from bool=2 to list=15), as that is the pattern used elsewhere.